### PR TITLE
GEODE-8990: Reworked out of date test case

### PR DESCRIPTION
**Pull request by BenjaminPerryRoss**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by rhoughton-pivot — APPROVED** [review]

File: geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/Jetty9CachingClientServerTest.java [hunk 1]
```diff
@@ -44,29 +45,37 @@ public ContainerInstall getInstall(IntSupplier portSupplier)
...
-    Client.Response resp = client.set(key, value);
+    Client.Response resp = client.set(key, localValue);
     assertThat(resp.getResponse()).isEqualTo("");
     assertThat(resp.getSessionCookie()).isNotEqualTo("");
 
+    // Make sure the client correctly set its original cached value
+    resp = client.get(key);
+    assertThat(resp.getResponse()).isEqualTo(localValue);
+
     serverVM.invoke("set bogus session key", () -> {
```
Comment (pivotal-eshu) [inline geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/Jetty9CachingClientServerTest.java:68]: Probably should change the runnable name here as the test expectation has been changed.

File: geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/Jetty9CachingClientServerTest.java [hunk 2]
```diff
@@ -44,29 +45,37 @@ public ContainerInstall getInstall(IntSupplier portSupplier)
...
 
-    // Make sure the client still sees its original cached value
-    resp = client.get(key);
-    assertThat(resp.getResponse()).isEqualTo(value);
+    // Make sure the client is now retrieving the value changed in the cluster
+    GeodeAwaitility.await().untilAsserted(() -> {
+      Client.Response response = client.get(key);
+      assertThat(response.getResponse()).isEqualTo(remoteValue);
+    });
+    // assertThat(resp.getResponse()).isEqualTo(remoteValue);
```
Comment (pivotal-eshu) [inline geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/Jetty9CachingClientServerTest.java:79]: The commented out code can be removed.

File: geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/Jetty9CachingClientServerTest.java (hunk 1 above)
Comment (BenjaminPerryRoss) [reply geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/Jetty9CachingClientServerTest.java:68]: Good idea! I've changed it to 'remote session key' instead.

File: geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/Jetty9CachingClientServerTest.java (hunk 2 above)
Comment (BenjaminPerryRoss) [reply geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/Jetty9CachingClientServerTest.java:79]: Done!

**Review by pivotal-eshu — APPROVED** [review]

**Review by onichols-pivotal — APPROVED** [review]

File: geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/session/tests/ServerContainer.java [hunk 3]
```diff
@@ -128,7 +128,7 @@ public ServerContainer(ContainerInstall install, File containerConfigHome,
     configuration.setProperty(GeneralPropertySet.LOGGING, loggingLevel);
     // Removes secureRandom generation so that container startup is much faster
     configuration.setProperty(GeneralPropertySet.JVMARGS,
-        "-Djava.security.egd=file:/dev/./urandom -Xmx128m -Xms64m");
+        "-Djava.security.egd=file:/dev/./urandom -Xmx256m -Xms128m");
```
Comment (sabbey37) [inline geode-assembly/geode-assembly-test/src/main/java/org/apache/geode/session/tests/ServerContainer.java:131]: Looks good and seems to pass consistently!  I think we can leave the `-Xms` at 64 though.

**Review by sabbey37 — APPROVED** [review]

**Comment by BenjaminPerryRoss** [comment]
The original solution present by this PR turned out to be inaccurate and was reverted. A new solution has been found, but for clarity (and to make sure we're not approving based on reviews from before the actually fix was implemented) I've created another PR here: https://github.com/apache/geode/pull/6443

**Pull request closed without merging**